source/server/server.cc

```diff
@@ -351,8 +351,9 @@ void InstanceImpl::initialize(const Options& options,
   hooks.onRuntimeCreated();
 
   // Once we have runtime we can initialize the SSL context manager.
-  ssl_context_manager_ =
-      std::make_unique<Extensions::TransportSockets::Tls::ContextManagerImpl>(time_source_);
+  auto& factory = Envoy::Config::Utility::getAndCheckFactory<Ssl::ContextManagerFactory>(
+      Ssl::ContextManagerFactory::name());
+  ssl_context_manager_ = factory.createContextManager(time_source_);
```

Comment:
What if there is no `Ssl::ContextManagerFactory`?

If you want to add the ability to compile Envoy without BoringSSL, that's fine, but then it should work without it, without adding undocumented dependency on some 3rd-party replacements.